### PR TITLE
Volunteer Routes - Still work in progress

### LINKED ISSUE #41: Create Volunteer Routes
Using our volunteer model, create a couple of routes for volunteers. Feel free to look at the event routes for reference. This should go in `/backend/routes/volunteer.js`: 

- [ ] GET /volunteer/:id - Method that returns a volunteer based on the id passed in. Throw an error if there is no volunteer associated with the id
- [ ] POST /volunteer/register- Method that adds a volunteer to MongoDB if their email is unique. Additionally, it adds the existing or new volunteer to the volunteers array in the event they are registering for.  All data should be passed into the body, including the eventId they are registering for. Throws an error if fields are missing, wrong, or the eventid doesn't exist. Return the volunteer that was registered as well as an isNew boolean indicating if this is a new or existing volunteer.

### PR DESCRIPTION
**Pull request by vboo123**
wanted to just push what I have, will finish it up in coming days

### CODE REVIEW

**Comment by vboo123** [comment]
Attempted to fix the volunteer routes. Let me know what needs to be changed

File: backend/routes/volunteer.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const express = require("express");
+const Volunteer = require("../models/volunteer");
+
+const router = express.Router();
+
+// #1 - get all volunteers
+router.get("/volunteer/:id", async (req, res) => {
+  try {
+    res.send({req})
```
Comment (vwinstea) [inline backend/routes/volunteer.js]: I'm unsure what you're doing here ... it looks like your just sending the request back. You should be querying the database using a [Mongoose query](https://mongoosejs.com/docs/queries.html). Lmk if you have any questions on how to implement it!

File: backend/routes/volunteer.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+router.post("/volunteer/register", async (req, res) => {
+  try {
+    const { firstName, lastName, email, phone, notes, eventID } = req.body;
+    let volunteer = new Volunteer({
+        firstName,
+        lastName,
+        email,
+        phone,
+        notes,
+        eventID
```
Comment (vwinstea) [inline backend/routes/volunteer.js]: The eventID isn't a property of the Volunteer, it's how we tie the event to the Volunteer. I'd read over the Issue description again and send me any questions you have.

**Review by vwinstea — CHANGES_REQUESTED** [review]
Hi Vboo, thanks for working on this some more! Unfortunately, there's a couple of issues with this PR that will prevent the routes from working. I'll message you on Slack and see what we can do to get this task done this week. I know learning backend code can be tricky, so please lmk at any point in the process if you have any questions at all!

**Comment by vboo123** [comment]
Final version here

**Review by vwinstea — APPROVED** [review]
Great work! FYI I updated the `volunteer/register` route so it accepts an eventID, so the data structure you need to pass in will look like this:
image

**Pull request merged**